feat: EPI-1291: update notes reference data

namhoangbes opened the pull request:
### Changes

_Add a brief description of the changes in this PR to help give the reviewer context._

### Deploys

- [x] **Deploy to Tamanu Internal** <!-- #deploy -->

### Tests

- [ ] **Run E2E Tests** <!-- #e2e -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

namhoangbes committed:
feat: EPI-1290: add system_required flag to reference data

namhoangbes committed:
fix: update tests and decription for systemRequired filed

namhoangbes committed:
fix: EPI-1290: move system required validation to import function

namhoangbes committed:
feat: EPI-1291: migrate note types to reference data

namhoangbes committed:
fix: EPI-1291: resolve failing test for note type reference data

namhoangbes committed:
fix: EPI-1291: migrate translation for note type

namhoangbes committed:
fix: EPI-1291: remove default translation creation

namhoangbes committed:
test: update test for translatable reference data importers

namhoangbes committed:
fix: update suggestion and filter notes by note type id

namhoangbes committed:
test: resolve failing test for filtering notes

namhoangbes committed:
Merge branch 'main' into feat/EPI-1290-add-system-required-flag-to-reference-data

namhoangbes committed:
fix: EPI-1291: set noteTypeId field when editing treatment plan note

namhoangbes committed:
fix: exclude noteType from provisioning validation

namhoangbes committed:
fix: update note type migration to reference data

khoi-bes commented on packages/database/src/migrations/1759894448776-migrateNoteTypesToReferenceData.ts:
```diff
@@ -0,0 +1,188 @@
+    },
+  );
+
+  await query.removeColumn('notes', 'note_type');
```
I think you forgot to remove the nullable prop below

namhoangbes committed:
Merge branch 'main' into feat/EPI-1291-update-notes-reference-data

namhoangbes committed:
fix: EPI-1291: update note template to use note type from reference data

khoi-bes committed:
Merge branch 'main' into feat/EPI-1291-update-notes-reference-data

khoi-bes committed:
Merge branch 'main' into feat/EPI-1291-update-notes-reference-data

khoi-bes committed:
refactor: Update NoteCommonFields and template components to improve code structure and remove unused Tooltip component

- Refactored imports in NoteCommonFields.jsx for better readability.
- Removed StyledTooltip and related logic from NoteCommonFields.jsx.
- Updated NewTemplateForm.jsx to use TranslatedEnum instead of TranslatedReferenceData for patient letter label.
- Simplified TemplateList.jsx by removing TranslatedReferenceData and using noteTypeNameMap for fallback names.
- Adjusted NotesPane.jsx to filter out specific note types in the suggester.

khoi-bes committed:
feat(migrations): Add note type migration and update note model

- Introduced a new migration to populate reference data for note types.
- Added a foreign key column `noteTypeId` to the notes table, linking to the reference data.
- Updated the Note model to reflect the new relationship with reference data for note types.
- Removed the old `noteType` column from the notes table and adjusted related queries accordingly.

namhoangbes committed:
fix: make note_type_id not nullable

namhoangbes committed:
fix: get translation for note type in template form

namhoangbes committed:
fix: display system note type in note form and hide in note filter

namhoangbes committed:
Merge branch 'main' into feat/EPI-1291-update-notes-reference-data

namhoangbes committed:
fix: resolve conflict in default-provisioning file

khoi-bes committed:
Merge branch 'main' into feat/EPI-1291-update-notes-reference-data

khoi-bes committed:
feat: update suggestions query to include noLimit option and refactor note type handling in forms

khoi-bes committed:
feat: add NoteType to ReferenceDataType for enhanced data handling

khoi-bes committed:
refactor: filter out SYSTEM note type from NoteTypeField and update note type filterer in NotesPane

khoi-bes committed:
Merge branch 'main' into feat/EPI-1291-update-notes-reference-data

khoi-bes committed:
feat(migrations): add updated_at_sync_tick to note type migration for reference data consistency

khoi-bes committed:
chore(migrations): remove obsolete migrateNoteTypesToReferenceData migration file

namhoangbes committed:
fix: add migrations to update note type for templates

namhoangbes committed:
refactor: remove unused interface and improve check note permission
